addons/l10n_es_edi_tbai/models/account_move.py

```diff
@@ -84,15 +89,30 @@ class AccountMove(models.Model):
     # API-DECORATED & EXTENDED METHODS
     # -------------------------------------------------------------------------
 
+    @api.depends('l10n_es_tbai_post_document_id.state', 'l10n_es_tbai_cancel_document_id.state')
+    def _compute_l10n_es_tbai_state(self):
+        for move in self:
+            state = 'to_send' if move.l10n_es_tbai_is_required else None
+            if move.l10n_es_tbai_post_document_id and move.l10n_es_tbai_post_document_id.state == 'accepted':
+                state = 'sent'
+            if move.l10n_es_tbai_cancel_document_id and move.l10n_es_tbai_cancel_document_id.state == 'accepted':
+                state = 'cancelled'
+
+            move.l10n_es_tbai_state = state
+
     @api.depends('move_type', 'company_id')
     def _compute_l10n_es_tbai_is_required(self):
         for move in self:
-            move.l10n_es_tbai_is_required = (move.is_sale_document() or move.is_purchase_document() and move.company_id.l10n_es_tbai_tax_agency == 'bizkaia'
-                                             and not any(t.l10n_es_type == 'ignore' for t in move.invoice_line_ids.tax_ids))\
-                and move.country_code == 'ES' \
-                and move.company_id.l10n_es_tbai_tax_agency
+            move.l10n_es_tbai_is_required = (
+                move.company_id.l10n_es_tbai_is_enabled
+                and (
+                    move.is_sale_document()
+                    or move.is_purchase_document() and move.company_id.l10n_es_tbai_tax_agency == 'bizkaia'
+                )
+                and any(not line._l10n_es_tbai_is_ignored() for line in move.invoice_line_ids)
```

Comment:
The problem with the flatten_tax_hierarchy added in that method is that the subtract base which is normally hidden in the group of taxes for the DUA, will make it ignore the line. We can simple change "ignore" to "" in the csv normally